File: mobile/lib/features/inventory/domain/entities/stock_transfer.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'stock_transfer.freezed.dart';
part 'stock_transfer.g.dart';

enum StockTransferType {
  @JsonValue('loading')
  loading,
  @JsonValue('unloading')
  unloading,
}

enum StockTransferStatus {
  @JsonValue('pending')
  pending,
  @JsonValue('confirmed')
  confirmed,
  @JsonValue('rejected')
  rejected,
}

@freezed
abstract class StockTransfer with _$StockTransfer {
  const factory StockTransfer({
    required String id,
    @JsonKey(name: 'warehouse_id') required String warehouseId,
    @JsonKey(name: 'sales_id') required String salesId,
    required StockTransferType type,
    required StockTransferStatus status,
    String? notes,
    @JsonKey(name: 'created_at') DateTime? createdAt,
    @JsonKey(name: 'updated_at') DateTime? updatedAt,
    @Default([]) List<StockTransferItem> items,
  }) = _StockTransfer;

  factory StockTransfer.fromJson(Map<String, dynamic> json) => _$StockTransferFromJson(json);
}

@freezed
abstract class StockTransferItem with _$StockTransferItem {
  const factory StockTransferItem({
    required String id,
    @JsonKey(name: 'transfer_id') required String transferId,
    @JsonKey(name: 'product_id') required String productId,
    required double quantity,
    @JsonKey(name: 'product_name') String? productName,
    @JsonKey(name: 'product_unit') String? productUnit,
  }) = _StockTransferItem;

  factory StockTransferItem.fromJson(Map<String, dynamic> json) => _$StockTransferItemFromJson(json);
}

File: backend/internal/domain/models/stock_transfer.go
package models

import (
	"crm_wowin_backend/pkg/utils"
	"github.com/google/uuid"
)

type StockTransferType string

const (
	StockTransferTypeLoading   StockTransferType = "loading"
	StockTransferTypeUnloading StockTransferType = "unloading"
)

type StockTransferStatus string

const (
	StockTransferStatusPending   StockTransferStatus = "pending"
	StockTransferStatusConfirmed StockTransferStatus = "confirmed"
	StockTransferStatusRejected  StockTransferStatus = "rejected"
)

type StockTransfer struct {
	ID          uuid.UUID           `json:"id"`
	WarehouseID uuid.UUID           `json:"warehouse_id"`
	SalesID     uuid.UUID           `json:"sales_id"`
	Type        StockTransferType   `json:"type"`
	Status      StockTransferStatus `json:"status"`
	Notes       *string             `json:"notes,omitempty"`
	CreatedAt   utils.FlexTime      `json:"created_at"`
	UpdatedAt   utils.FlexTime      `json:"updated_at"`
	
	Items []*StockTransferItem `json:"items,omitempty"`
}

type StockTransferItem struct {
	ID         uuid.UUID `json:"id"`
	TransferID uuid.UUID `json:"transfer_id"`
	ProductID  uuid.UUID `json:"product_id"`
	Quantity   float64   `json:"quantity"`
    CreatedAt  utils.FlexTime `json:"created_at"`
}
